pulp_service/pulp_service/app/authorization.py

```diff
@@ -23,33 +26,52 @@ def has_permission(self, request, view):
         # Admins have all permissions
         if request.user.is_superuser:
             return True
-        # Decode the identity header
+        user = request.user
+
+        # The user that created the domain has permission to access that domain
+        if DomainOrg.objects.filter(domain=get_domain(), user=user).exists():
```

Comment:
I agree with what you said, it would be better to combine this with the `filter` further down. When combining use a Q() filter also so that it only goes to the DB once instead of two `filter()` calls combined with a Python `or`.